File: Maximize Stock/maximizeprice.py
class Solution(object):
    def maxProfit(self, prices):
        if len(prices) == 0: return 0
        else:
            max_profit = 0
            min_price = prices[0]
            for sell in range(len(prices)):
                    min_price=min(min_price,prices[sell])
                    max_profit = max(prices[sell] - min_price, max_profit)
            return max_profit

File: Maximize Stock/maximizeprice.ts
function maxProfit(prices: number[]): number {
    if (!prices || prices.length === 1) {
        return 0;
    }
    let min = prices [0];
    let max = 0;
    for (const price of prices) {
        min = Math.min(min, price);
        const potentialPrices = price - min;
        max = Math.max (max, potentialPrices);
    }
    return max;
}